### PR TITLE
Switch to the "Swift" diagnostic style as the default

### PR DESCRIPTION
**Pull request by DougGregor**
The new "Swift" diagnostic style has been enabled for standard library builds for quite a while now, and we've addressed all of the issues we've seen with it. Turn it on by default.

There are a significant number of FileCheck-based tests in our test suite that are checking for the precise output of the LLVM diagnostic formatter, and do not pass with the new Swift one. Generalize those tests that are easy to generalize, so they pass with both, and force others to use the older formatter with `-diagnostic-style llvm`. If we need to switch back, we can leave the tests and switch the default back easily enough.

This resolves one old compiler crasher due to some poor Unicode handling in the older diagnostic formatter.

### CODE REVIEW

**Comment by DougGregor** [comment]
https://github.com/apple/swift-syntax/pull/2501

@swift-ci please test

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7349
https://github.com/apple/swift-syntax/pull/2501

@swift-ci please test

**Review by Azoy — APPROVED** [review]

**Comment by DougGregor** [comment]
Python on Windows seems unhappy with some of the Unicode we are producing:
```
UnicodeEncodeError: 'charmap' codec can't encode character '\u2502' in position 219: character maps to <undefined>
```

File: test/NameLookup/scope_map_lookup_extension_extension.swift [hunk 1]
```diff
@@ -2,5 +2,12 @@
 // RUN: not %target-swift-frontend -typecheck %s 2> %t.errors
 // RUN: %FileCheck %s <%t.errors
 // CHECK-NOT: Program arguments:
+
+// Note: extra newlines below ensure that context printing doesn't show the
```
Comment (ahoppen) [inline test/NameLookup/scope_map_lookup_extension_extension.swift:6]: I think this comment would be useful in the other tests that have multiple newlines as well. I was quite confused by them at first.

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7349

@swift-ci please test

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7368

@swift-ci please test

**Comment by slavapestov** [comment]
@DougGregor Hahaha you finally fixed that compiler_crasher! Nice! Only one left...

**Pull request merged**